ci/main.go

```diff
@@ -0,0 +1,211 @@
+			$(GO_TO_HOST) build $(EXTRA_HOST_ARGS) -o $@ github.com/iansmith/parigot/command/protoc-gen-parigot
+	*/
+
+	var err error
+	// template, err = findFilesEndWith("command/protoc-gen-parigot", ".tmpl")
+	// if err != nil {
+	// 	return img, err
+	// }
+	// eneratorSrc, err = findFilesEndWith("test", ".go")
+	// if err != nil {
+	// 	return img, err
+	// }
+	template, err = img.WithExec([]string{"bash", "-c", `find command/protoc-gen-parigot -type f -regex ".*\.tmpl"`}).Stdout(ctx)
+	if err != nil {
+		return img, err
+	}
+	eneratorSrc, err = img.WithExec([]string{"bash", "-c", `find command/protoc-gen-parigot -type f -regex ".*\.go"`}).Stdout(ctx)
+	if err != nil {
+		return img, err
+	}
+
+	// contents, err := img.WithExec([]string{"ls", "-l", "."}).Stdout(ctx)
+	// if err != nil {
+	// 	return img, err
+	// }
+	// fmt.Println(contents)
+
+	target := "build/protoc-gen-parigot"
+	packagePath := "github.com/iansmith/parigot/command/protoc-gen-parigot"
+	img = img.WithExec([]string{"rm", "-f", target})
+	img = img.WithExec([]string{GO2HOST, "build", "-o", target, packagePath})
+
+	return img, nil
+}
+
+func generateRep(ctx context.Context, img *dagger.Container) (*dagger.Container, error) {
+	/*
+		API_PROTO=$(shell find api/proto -type f -regex ".*\.proto")
+		TEST_PROTO=$(shell find test -type f -regex ".*\.proto")
+
+		## we just use a single representative file for all the generated code from
+		REP=g/file/$(API_VERSION)/file.pb.go
+		$(REP): $(API_PROTO) $(TEST_PROTO) build/protoc-gen-parigot
+			@rm -rf g/*
+			buf lint
+			buf generate
+	*/
+
+	var err error
+	apiProto, err = img.WithExec([]string{"bash", "-c", `find api/proto -type f -regex ".*\.proto"`}).Stdout(ctx)
+	if err != nil {
+		return img, err
+	}
+	testProto, err = img.WithExec([]string{"bash", "-c", `find test -type f -regex ".*\.proto"`}).Stdout(ctx)
+	if err != nil {
+		return img, err
+	}
+
+	img = img.WithExec([]string{"rm", "-rf", "g/*"})
+	img = img.WithExec([]string{"buf", "lint"})
+	img = img.WithExec([]string{"buf", "generate"})
+
+	return img, nil
+}
+
+func buildFilePWasm(ctx context.Context, img *dagger.Container) (*dagger.Container, error) {
+	/*
+		// client side of the file service
+		FILE_SERVICE=$(shell find apiwasm/file -type f -regex ".*\.go")
+		build/file.p.wasm: $(FILE_SERVICE) $(REP) $(SYSCALL_CLIENT_SIDE)
+			@rm -f $@
+			$(GO_TO_WASM) build  $(EXTRA_WASM_COMP_ARGS) -tags "buildvcs=false" -o $@ github.com/iansmith/parigot/apiwasm/file
+	*/
+	var err error
+	_, err = img.WithExec([]string{"bash", "-c", `find apiwasm/file -type f -regex ".*\.go"`}).Stdout(ctx)
+	if err != nil {
+		return img, err
+	}
+
+	target := "build/file.p.wasm"
+	packagePath := "github.com/iansmith/parigot/apiwasm/file"
+	img = img.WithExec([]string{"rm", "-f", target})
+	img = img.WithExec([]string{Go2WASM, "build", "-tags", `"buildvcs=false"`, "-o", target, packagePath})
+
+	return img, nil
+}
+
+// func findFilesEndWith(path string, suffix string) ([]string, error) {
```

Comment:
OOOh now I find the code I was guessing was already in dagger!

😆

Reply:
You are partially correct. This function is designed to recursively find files with a specific suffix. However, it uses a package called filepath. Unfortunately, I didn't find any useful API in dagger :(

I feel like it is much easier to use the shell cmd than this function.

Reply:
Seems very odd to me that the dagger folks didn't have some functions built-in to find files.  But, if they don't you can do it with file path and walking the directories.  There are some interesting answers here:


https://stackoverflow.com/questions/47266529/go-find-files-in-directory-recursively

Reply:
Yes, that's exactly what I tried before, but I think cmd is more concise. Do you want me to switch to the filepath?